metrics/provider.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	metricsNamespace = "mattermod"
+	httpNamespace    = "requests"
+	cronNamespace    = "cron"
+	githubNamespace  = "github"
+
+	defaultPrometheusTimeoutSeconds = 60
+)
+
+// Provider is the interface that exposes the communication with the metrics system
+// this interface should be implemented by the different providers we want to include
+type Provider interface {
+	// ObserverHTTPRequestDuration stores the elapsed time for an HTTP request
```

Comment:
Should we use RTT instead?
```suggestion
	// ObserverHTTPRequestDuration stores the round-trip time for an HTTP request
```